Objects/exceptions.c

```diff
@@ -206,6 +343,26 @@ BaseException_set_args(PyBaseExceptionObject *self, PyObject *val, void *Py_UNUS
     return 0;
 }
 
+static PyObject *
+BaseException_get_kwargs(PyBaseExceptionObject *self, void *Py_UNUSED(ignored)) {
+    if (self->kwargs == NULL) {
+        self->kwargs = PyDict_New();
+    }
+    Py_XINCREF(self->kwargs);
+    return self->kwargs;
+}
+
+static int
+BaseException_set_kwargs(PyBaseExceptionObject *self, PyObject *val, void *Py_UNUSED(ignored)) {
+    if (val == NULL) {
+        PyErr_SetString(PyExc_TypeError, "kwargs may not be deleted");
+        return -1;
+    }
```

Comment:
I thought about doing that but it seems to me that `PyMapping_Check` is only a loose check, for example, wouldn't we be able to do:
```
>>> b = BaseException()
>>> b.kwargs = ()
```
since tuples support slicing?

If so, is the check still worth it?